Add support for WebGL2 FramebufferTextureLayer

mmatyas opened the pull request:
Adds support for `FramebufferTextureLayer` WebGL2 call.

See: https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.4

<!-- Please describe your changes on the following line: -->

Depends on #25798.

cc @jdm @zakorgy 

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

mmatyas committed:
Add support for WebGL2 FramebufferTextureLayer

Adds support for `FramebufferTextureLayer` WebGL2 call.

See: https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.4

highfive commented:
Heads up! This PR modifies the following files:
 * @asajeffrey: components/script/dom/webglrenderingcontext.rs, components/script/dom/webidls/WebGL2RenderingContext.webidl, components/script/dom/webglframebuffer.rs, components/script/dom/webgl2renderingcontext.rs, components/script/dom/webglrenderbuffer.rs
 * @jgraham: tests/wpt/webgl/meta/conformance2/renderbuffers/framebuffer-test.html.ini, tests/wpt/webgl/meta/conformance2/context/methods-2.html.ini, tests/wpt/webgl/meta/conformance2/rendering/framebuffer-texture-level1.html.ini, tests/wpt/webgl/meta/conformance2/rendering/framebuffer-completeness-unaffected.html.ini, tests/wpt/webgl/meta/conformance2/renderbuffers/framebuffer-object-attachment.html.ini and 3 more
 * @kichjang: components/script/dom/webglrenderingcontext.rs, components/script/dom/webidls/WebGL2RenderingContext.webidl, components/script/dom/webglframebuffer.rs, components/script/dom/webgl2renderingcontext.rs, components/script/dom/webglrenderbuffer.rs

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!

Manishearth commented:
r? @nox or @jdm for webgl stuff

bors-servo commented:
:umbrella: The latest upstream changes (presumably #25784) made this pull request unmergeable. Please resolve the merge conflicts.

mmatyas commented:
Rebased and updated the tests. There aren't too many of them though, as this GL call mainly depends on Texture2D/3D implemented first, but the code itself is ready for review.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #25798) made this pull request unmergeable. Please resolve the merge conflicts.

mmatyas commented:
Rebased.

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 8389189 has been approved by `jdm`

<!-- @bors-servo r=jdm 8389189d943dfd5b404401ae41e73da71ec774be -->

bors-servo commented:
:hourglass: Testing commit 8389189d943dfd5b404401ae41e73da71ec774be with merge b0324f7d7ce77089fb48d90e43f2270b0e91336d...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/OHzeA4Z7Rk6uE4ewpKy1UQ)

CYBAI commented:
@bors-servo retry

- #24622, cygwin oom

bors-servo commented:
:hourglass: Testing commit 8389189d943dfd5b404401ae41e73da71ec774be with merge 301980c840237b19c609eb4155f15b25a5928b04...

bors-servo commented:
:sunny: Test successful - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/L1nsNPIpSwiPJnuQujA6MA)
Approved by: jdm
Pushing 301980c840237b19c609eb4155f15b25a5928b04 to master...
